### DISCUSSION TITLE
Big files support

### DISCUSSION THREAD

**Discussion by koxo** [DirectoryLister/DirectoryLister#580] [Help & Support]
Hi,

I have a problem with big files, like ISO. (5GB)

With the SplFileInfo::getSize():

How to cortrect this problem ? 

Thank you for your job.

**Comment by PHLAK** [comment]
Please follow the instructions in the [Troubleshooting docs](https://docs.directorylister.com/help-and-support/troubleshooting) and if you continue to have an issue please share additional details.

**Reply by koxo** [reply]
I have check through debug :  

Twig \ Error \ RuntimeError
An exception has been thrown during the rendering of a template ("SplFileInfo::getSize(): stat failed for ./my.ISO").
Previous exceptions
SplFileInfo::getSize(): stat failed for ./my.ISO (0)

this file is more than 4gb, so it must use a 64 bit var.

how to handle this

best regards.

**Reply by PHLAK** [reply]
Thanks for the information. I'll look into this.

**Reply by PHLAK** [reply]
Actually, I'm not sure this is an integer issue. Are you sure this file isn't corrupt? It seems like it's having trouble getting the file information in the first place.

**Reply by koxo** [reply]
Hi, maybe not, but the file is good it's just an ISO file of windows 10. if you want i can give you the link (private).

the problem is maybe here : 

php
 
namespace App\ViewFunctions;
 
use Symfony\Component\Finder\SplFileInfo;
 
class SizeForHumans extends ViewFunction
{
    /** @var string The function name */
    protected $name = 'size_for_humans';
 
    /** Get the human readable file size from a file object. */
    public function __invoke(SplFileInfo $file): string
    {
        $sizes = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
        $factor = (int) floor((strlen((string) $file->getSize()) - 1) / 3);
 
        return sprintf('%.2f', $file->getSize() / pow(1024, $factor)) . $sizes[$factor];
    }
}

**Reply by koxo** [reply]
"SplFileInfo::getSize(): stat failed for ./MY.ISO"

**Reply by koxo** [reply]
https://www.php.net/manual/fr/splfileinfo.getsize.php

**Reply by koxo** [reply]
nobody have tested big files (>2gb) ?
the function getSize seems to return only int ?

**Comment by PHLAK** [comment]
I recently confirmed that Directory Lister was able to list a 5 GB file on a 64 bit Linux system without issue.

**Reply by koxo** [reply]
Hi Chris, i can send you an url with a folder with big files and debug on, if you want to see.

**Reply by PHLAK** [reply]
Not sure how that would help any more than the error you already posted but you're welcome to send it.
